src/components/PreviewSection/PreviewSection.tsx

```diff
@@ -9,26 +12,49 @@ import {
   Name,
   Wrapper,
 } from "./PreviewSection.styles";
-import { IPreviewCard } from "@/utils/interfaces";
-import PreviewCard from "@/components/PreviewCard/PreviewCard";
-import Carousel from "../Carousel/Carousel";
 
-const PreviewSection = ({ cards }: { cards: { fields: IPreviewCard }[] }) => {
+const PreviewSection = ({ cards }: { cards: IPreviewCards[] }) => {
+  const [selectedRegion, setSelectedRegion] = useState("Nordeste");
+  const [selectedState, setSelectedState] = useState("");
+
+  const allCardsData = cards.map((card) => card.fields.jsonFile);
+
+  const filteredCards = allCardsData.map((regionData) => {
```

Comment:
Would be better to use a [useMemo](https://react.dev/reference/react/useMemo) here, I think. Then everytime the selecedRegion is updated the filteredCards are updated